sdk-utils/src/main/kotlin/com/rakuten/tech/mobile/sdkutils/eventlogger/Event.kt

```diff
@@ -19,32 +19,39 @@ internal data class Event(
     val rmcSdks: Map<String, String>? = null,
     val info: Map<String, String>? = null,
     var occurrenceCount: Int = 0,
-    @SerializedName("firstOccurrenceOn") var firstOccurrenceMillis: Long? = null
-) {
-    val eventVersion = "1.0"
-    val platform = Platform.ANDROID.displayName
-}
+    val eventVersion: String = "1.0",
+    val platform: String = "Android",
+    @SerializedName("firstOccurrenceOn") val createdOn: Long = System.currentTimeMillis()
+)
 
 // ---------------------------- Event enums ----------------------------
 internal enum class EventType(val displayName: String) {
     CRITICAL("0"),
     WARNING("1")
 }
 
-internal enum class Platform(val displayName: String) {
-    ANDROID("Android")
-}
-
 // ---------------------------- Event extensions ----------------------------
 /**
  * Returns the string hash based on some data from the [Event].
  */
-internal fun Event.getIdentifier() = "${this.appVersion}${this.sdkName}${this.errorCode}${this.errorMessage}"
+internal fun Event.generateEventIdentifier() = generateEventIdentifier(
+    eventType,
+    appVersion,
+    sdkName,
+    errorCode,
+    errorMessage
+)
+
+internal fun generateEventIdentifier(
+    eventType: String,
+    appVersion: String,
+    sdkName: String,
+    errorCode: String,
+    errorMessage: String
+) = "$eventType$appVersion$sdkName$errorCode$errorMessage"
     .getMD5HashData()
```

Comment:
do we know what may be the max length of string return here ?

Reply:
I am afraid if it may return a very large string

Reply:
MD5 hash is always 32 characters long

Reply:
Okay, thank you.